From 1762116125ee25d2a2f8c517f6071ca79f8814db Mon Sep 17 00:00:00 2001
From: Eric Chanudet <chanudete@ainfosec.com>
Date: Thu, 9 Aug 2018 16:26:54 -0400
Subject: [PATCH 10/22] config: Allow build system integration.

Openembedded will redefined build tools to match the staging
environment. Remove lazy assignements and replace with default
assignements.

Signed-off-by: Eric Chanudet <chanudete@ainfosec.com>
---
 Config.mk | 28 ++++++++++++----------------
 1 file changed, 12 insertions(+), 16 deletions(-)

diff --git a/Config.mk b/Config.mk
index bc6c8d4..d1a6604 100644
--- a/Config.mk
+++ b/Config.mk
@@ -45,22 +45,18 @@ CFLAGS_WARN       = -Wall -Wformat-security -Werror -Wstrict-prototypes \
 	            $(call cc-option,$(CC),-Wlogical-op,) \
 	            -Wno-missing-field-initializers
 
-AS         = as
-LD         = ld
-CC         = gcc
-CPP        = cpp
-AR         = ar
-RANLIB     = ranlib
-NM         = nm
-STRIP      = strip
-OBJCOPY    = objcopy
-OBJDUMP    = objdump
-
-ifeq ($(debug),n)
-INSTALL_STRIP = -s
-endif
-
-INSTALL      = install
+AS         ?= as
+LD         ?= ld
+CC         ?= gcc
+CPP        ?= cpp
+AR         ?= ar
+RANLIB     ?= ranlib
+NM         ?= nm
+STRIP      ?= strip
+OBJCOPY    ?= objcopy
+OBJDUMP    ?= objdump
+
+INSTALL      ?= install
 INSTALL_DIR  = $(INSTALL) -d -m0755 -p
 INSTALL_DATA = $(INSTALL) -m0644 -p
 INSTALL_PROG = $(INSTALL) $(INSTALL_STRIP) -m0755 -p
-- 
2.16.1

